pydantic_ai_slim/pydantic_ai/toolsets/searchable.py

```diff
@@ -0,0 +1,128 @@
+import logging
+import re
+from collections.abc import Callable
+from dataclasses import dataclass, field, replace
+from typing import Any, TypedDict
+
+from pydantic import TypeAdapter
+from typing_extensions import Self
+
+from .._run_context import AgentDepsT, RunContext
+from ..tools import ToolDefinition
+from .abstract import AbstractToolset, SchemaValidatorProt, ToolsetTool
+
+_SEARCH_TOOL_NAME = 'load_tools'
+
+
+class _SearchToolArgs(TypedDict):
+    regex: str
+
+
+def _search_tool_def() -> ToolDefinition:
```

Comment:
Check out `Tool.from_schema` and the `Tool` constructor that takes a function (as used by `FunctionToolset`) for easier ways to construct a single tool. The function approach is the easiest by far